src/main/java/uk/ac/ebi/atlas/experimentpage/tabs/ExperimentPageContentService.java

```diff
@@ -144,7 +145,18 @@ public JsonArray getPerplexities(String experimentAccession) {
 
     public JsonArray getMetadata(String experimentAccession) {
         var metadataArray = new JsonArray();
-        cellMetadataService.getMetadataTypes(experimentAccession)
+        var metadataTypes = cellMetadataService
+                .getMetadataTypes(experimentAccession)
+                .stream()
+                .filter(type -> cellMetadataService
+                        .getMetadataValuesForGivenType(experimentAccession, type)
+                        .values()
+                        .stream()
+                        .distinct()
+                        .count() >= 2)//we only need to show those metadata type which have more than 2 unique values
+                .collect(ImmutableSet.toImmutableSet());
+
+        metadataTypes
```

Comment:
I think we could remove the declaration for the new variable `metadataTypes` by merging the stream calculation.

Reply:
did it!